packages/cozy-harvest-lib/test/components/KonnectorModal.spec.js

```diff
@@ -0,0 +1,35 @@
+/* eslint-env jest */
+import React from 'react'
+import { shallow } from 'enzyme'
+
+import { KonnectorModal } from 'components/KonnectorModal'
+import TriggerManager from 'components/TriggerManager'
+
+const t = jest.fn().mockImplementation(key => key)
+
+const findAccountMock = jest.fn().mockImplementation(() => ({
+  id: 123,
+  doctype: 'io.cozy.accounts'
+}))
+
+describe('KonnectorModal', () => {
+  it('should pass trigger if konnector has triggers', async () => {
+    const mockKonnector = {
+      slug: 'mock',
+      name: 'Mock',
+      triggers: {
+        data: [{ id: 784, doctype: 'io.cozy.triggers' }]
+      }
+    }
+    const props = {
+      findAccount: findAccountMock,
+      dismissAction: jest.fn(),
+      onSuccess: jest.fn(),
+      konnector: mockKonnector,
+      t
+    }
+    const component = await shallow(<KonnectorModal {...props} />)
```

Comment:
I never used `await` for `shallow`, what does it do ?

Reply:
I dont think it is correct.

Reply:
It will wait for async `componentDidMount`, here we fetch the account after mounting. If I removed it, I'll get the loading status here

Reply:
oh, I did not know it was possible to do that. Nice.

Reply:
`componentDidMount` is not asynchronous here (only `fetchAccount` is).

First solution I see is:
```
componentDidMount(){
 this.fetchAccountPromise = this.fetchAccount()
}
```

And in test:
```
  const wrapper = shallow(<KonnectorModal {...props} />)
  await wrapper.instance().fetchAccountPromise
```

Reply:
Seems good enough